homeassistant/components/iammeter/const.py

```diff
@@ -0,0 +1,219 @@
+
+    value: Callable[[float | int], float] | Callable[[datetime], datetime] | None = None
+
+
+SENSOR_TYPES_3080: tuple[IammeterSensorEntityDescription, ...] = (
+    IammeterSensorEntityDescription(
+        key="Voltage",
+        name="Voltage",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    IammeterSensorEntityDescription(
+        key="Current",
+        name="Current",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    IammeterSensorEntityDescription(
+        key="Power",
+        name="Power",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    IammeterSensorEntityDescription(
+        key="ImportEnergy",
+        name="ImportEnergy",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    IammeterSensorEntityDescription(
+        key="ExportGrid",
+        name="ExportGrid",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+SENSOR_TYPES_3080T: tuple[IammeterSensorEntityDescription, ...] = (
+    IammeterSensorEntityDescription(
+        key="Voltage_A",
+        name="Voltage_A",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    IammeterSensorEntityDescription(
+        key="Current_A",
+        name="Current_A",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    IammeterSensorEntityDescription(
+        key="Power_A",
+        name="Power_A",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    IammeterSensorEntityDescription(
+        key="ImportEnergy_A",
+        name="ImportEnergy_A",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    IammeterSensorEntityDescription(
+        key="ExportGrid_A",
+        name="ExportGrid_A",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    IammeterSensorEntityDescription(
+        key="Frequency_A",
+        name="Frequency_A",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=UnitOfFrequency.HERTZ,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    IammeterSensorEntityDescription(
```

Comment:
- Device class missing
- Entity category missing
- We should disable this sensor by default